docs/source/common/lightning_module.rst

```diff
@@ -1064,7 +1064,9 @@ override :meth:`pytorch_lightning.core.LightningModule.tbptt_split_batch`:
 
 Hooks
 ^^^^^
-This is the pseudocode to describe how all the hooks are called during a call to ``.fit()``.
+This is the pseudocode to describe the structure of :meth:`~pytorch_lightning.trainer.Trainer.fit`.
+The inputs and outputs of each function are not represented for simplicity. Please check each function's API reference
```

Comment:
I prefer keeping `outputs` here since its very helpful to distinguish some hooks (`on_train_epoch_end` vs `training_epoch_end`) and since inputs/outputs across hooks are closely related to the order of execution

Reply:
I removed it because I find it weird having the argument for `training_step` but not all the other functions that take a `batch` argument or an `outputs` argument.

So either I would add the batch argument for all functions or for none of them.

But adding it for all decreases the simplicity

Reply:
that's fair. should we have a separate section to discuss how data is passed across hooks (not for this PR)?
it can clarify https://github.com/PyTorchLightning/pytorch-lightning/pull/7723

Reply:
Sure. If you have any ideas on how to clear it do not hesitate to send a patch or open an issue